### PR TITLE
Feat(eos_cli_config_gen): Add support for IPv6 Helper Addresses on VLAN interfaces

### LINKED ISSUE #3010: Feat(eos_cli_config_gen): Add support for IPv6 Helper Addresses on VLAN interfaces
### Enhancement summary

Add support for configuring ipv6_helper addresses on VLAN interfaces.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

It may be necessary to add IPv6 helper addresses in campus networks that use IPv6.

### Describe the solution you would like

Add the `ipv6_helpers` key and values to the supported keys under **eos_cli_config_gen's** `vlan_interfaces` data model.

```
vlan_interfaces:
  - name: <str>
    ipv6_helpers:
      -  ip_helper: <str>
         link_address: <str>
         local_interface: <str>
         source_address: <str>
         vrf: <str>
```

### Describe alternatives you have considered

The `eos_cli` key in the `vlan_interfaces` dictionary is a current way to configure this.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KyleMui**
## Change Summary

Add interface configurations to have ipv6_helper compatibility

## Related Issue(s)

Fixes #3010

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Added the below to the eos_cli_config_gen `vlan_interfaces` data model
```
- name: <str>
  shutdown: false
  ipv6_address_virtual: <str>
  ipv6_helpers:
    - ipv6_helper: <str>
      vrf: <str>
      source_address: <str>
      local_interface: <str>
      link_address: <str>
```

## How to test

Added to the eos_cli_config_gen molecule scenario `vlan_interfaces` host-var
```
- name: Vlan334
    description: SVI description
    shutdown: false
    ipv6_address_virtual: 2001:db9::1/64
    ipv6_helpers:
      - ipv6_helper: 2001:dbs8::aaaa
        vrf: A
        source_address: 2001:db8::aaab
      - ipv6_helper: 2001:db8::bbbb
        vrf: A
        local_interface: Loopback0
      - ipv6_helper: 2001:db8::cccc
        vrf: A
        source_address: 2001:db8::cccd
        link_address: 2001:db8::ccce
      - ipv6_helper: 2001:db8::dddd
        link_address: 2001:db8::ddde
      - ipv6_helper: 2001:db8::eeee
        vrf: A
        local_interface: Loopback0
        link_address: 2001:db8::eeef
```
renders to:
```
interface Vlan334
   description SVI description
   no shutdown
   ipv6 address virtual 2001:db9::1/64
   ipv6 helper-address 2001:db8::aaaa vrf A source-address 2001:db8::aaab
   ipv6 helper-address 2001:db8::bbbb vrf A local-interface Loopback0
   ipv6 helper-address 2001:db8::cccc vrf A source-address 2001:db8::cccd link-address 2001:db8::ccce
   ipv6 helper-address 2001:db8::dddd link-address 2001:db8::ddde
   ipv6 helper-address 2001:db8::eeee vrf A local-interface Loopback0 link-address 2001:db8::eeef
```
which has been verified as valid config on a veos node

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vlan_interfaces.schema.yml [hunk 1]
```diff
@@ -94,6 +94,30 @@ keys:
                 description: VRF where DHCP server can be reached
                 convert_types:
                   - int
+        ipv6_helpers:
```
Comment (jrecchia1029) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vlan_interfaces.schema.yml]: Need to move this key down so it is in order with the other ipv6 keys

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2 [hunk 2]
```diff
@@ -164,6 +164,21 @@ interface {{ vlan_interface.name }}
 {%     if vlan_interface.ipv6_access_group_out is arista.avd.defined %}
    ipv6 access-group {{ vlan_interface.ipv6_access_group_out }} out
 {%     endif %}
+{%     for ipv6_helper in vlan_interface.ipv6_helpers | arista.avd.natural_sort('ipv6_helper') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:183]: I have not checked, but please verify that the order of the ipv6 helpers match the config ordering on EOS. (Both general order of commands on the SVI, but also if the order of individual helpers are sorted similar to natural_sort)

Comment (KyleMui) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2:183]: In eos ipv6 helper addresses with no vrf defined are listed before those with a vrf defined.  To account for this I created two for loops which go through the ipv6 helper addresses. The first one checking for those without a vrf and the second one checking for those with a vrf. I think this is the best way to do this when factoring in simplicity and cost. Let me know if you have any other suggestions.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vlan_interfaces.schema.yml (hunk 1 above)
Comment (KyleMui) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vlan_interfaces.schema.yml]: done

**Review by jrecchia1029 — APPROVED** [review]
LGTM

**Review by gmuloc — COMMENTED** [review]
So quick question on this one

when I typed in my CLI (EOS 4.27)

```
Spine1(config-if-Vl42)#ipv6 helper-address dead::cafe link-address fe80::1234
Spine1(config-if-Vl42)#show active
interface Vlan42
   ipv6 dhcp relay destination dead::cafe link-address fe80::1234
```

the command is translated to dhcp relay. Looking at some internal doc, it looks like the second syntax was introduced later and we are still supporting the helper-address syntax for convenience. The other syntax is already supported by eos_cli_config_gen.

@KyleMui - can you please let us know which EOS version you have been testing on?

I am not sure we want to be rendering configuration that is not going to show up in the eos config in the end (except maybe for older sofrware versions ? In that cas I would rather have something in the schema stating so)

**Comment by KyleMui** [comment]
> So quick question on this one
> 
> when I typed in my CLI (EOS 4.27)
> 
> ```
> Spine1(config-if-Vl42)#ipv6 helper-address dead::cafe link-address fe80::1234
> Spine1(config-if-Vl42)#show active
> interface Vlan42
>    ipv6 dhcp relay destination dead::cafe link-address fe80::1234
> ```
> 
> the command is translated to dhcp relay. Looking at some internal doc, it looks like the second syntax was introduced later and we are still supporting the helper-address syntax for convenience. The other syntax is already supported by eos_cli_config_gen.
> 
> @KyleMui - can you please let us know which EOS version you have been testing on?
> 
> I am not sure we want to be rendering configuration that is not going to show up in the eos config in the end (except maybe for older sofrware versions ? In that cas I would rather have something in the schema stating so)


I tested the config onto EOS version 4.30.0F and it also translates to `ipv6 dhcp relay destination`.  I think we can close the PR and issue since the `eos_cli_config_gen` already supports this command.

**Pull request closed without merging**